src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting_NetFx/WebTools.cs

```diff
@@ -240,8 +242,27 @@ private static MethodInfo HandleRequestAsyncMethod
             BufferManager bufferManager,
             ILspLogger logger)
         {
-            var instance = CreateInstance(Type, textBufferFactoryService, bufferManager.Instance, logger);
+            var instance = CreateInstance(Type, textBufferFactoryService, bufferManager.Instance, new LegacyClaspILspLogger(logger));
             return new(instance);
         }
+
+        /// <summary>
+        /// Wraps the razor logger (from the clasp source package) into the binary clasp logger that webtools uses.
+        /// </summary>
+        /// <param name="logger"></param>
+        private class LegacyClaspILspLogger(ILspLogger logger) : LegacyClasp.Microsoft.CommonLanguageServerProtocol.Framework.ILspLogger
```

Comment:
I'm guessing that this is necessary until Web Tools is updated and then we pull in new Web Tools assemblies for tests? If so, could you please be sure to file an issue for us so we remember to get back to this?

Reply:
Yeah we can remove the code that ships the legacy clasp package.  I filed https://github.com/dotnet/razor/issues/10160 for that.

However I expect this particular bit of code will actually need to change to a reflection instantiation of the ILspLogger from the webtools package (since the razor source package type isn't the same type as the webtools source package type).  Its unfortunate that the clasp type is exposed across boundaries here.

Reply:
Yeah, that makes sense. Of course, we use so much reflection for the WebTools code here that's really just business as usual. :smile: